src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1727,6 +1770,12 @@ protected override void InvokeDefaultAction(string path)
                                 return;
                             }
 
+                            //If symlink points to its parent directory then do not pursue symlink further
+                            if (IsRecursiveSymLink(recursiveDirectory))
+                            {
+                                    recurse=false;
+                                    return;
+                            }
```

Comment:
If the current directory is recursive, I think we should just skip this one and carry on with other directories. Return back without processing the rest of directories doesn't look right.